libtsuba/src/AddProperties.cpp

```diff
@@ -20,18 +20,11 @@ katana::Result<std::shared_ptr<arrow::Table>>
 DoLoadProperties(
     const std::string& expected_name, const katana::Uri& file_path,
     std::optional<tsuba::ParquetReader::Slice> slice = std::nullopt) {
-  auto reader_res = tsuba::ParquetReader::Make();
-  if (!reader_res) {
-    return reader_res.error().WithContext("loading property");
-  }
-  std::unique_ptr<tsuba::ParquetReader> reader = std::move(reader_res.value());
-
-  auto out_res = reader->ReadTable(file_path, slice);
-  if (!out_res) {
-    return out_res.error().WithContext("loading property");
-  }
+  auto reader =
+      KATANA_CHECKED_CONTEXT(tsuba::ParquetReader::Make(), "loading property");
```

Comment:
We might take this opportunity to reduce stuttering in errors. I think a good rule of thumb is not to report context that the caller knows. In this case I don't think "loading property" is adding anything that the caller of this doesn't know.

Reply:
+1 
Though personally I do prefer we lean towards over-providing context rather than under-providing it.

Reply:
Agreed. There are certain places I removed it where it didn't provide much value, but in general I think adding some context can be helpful when looking at an error message. Unless @tylershunt you think it's confusing and should only be there if it provides needed context? E.g. if it's just a isolated function call, there may not be any context needed since the error gives the line number?

Reply:
I think `KATANA_CHECKED` is easier to read and should be preferred.  `KATANA_CHECKED_CONTEXT` is for cases where the context, beyond file and line number, is valuable and (always?) that includes data values.